src/main/java/randoop/generation/OrienteeringSelection.java

```diff
@@ -0,0 +1,244 @@
+  private final Map<Sequence, Integer> sequenceSelectionCount = new HashMap<>();
+
+  /**
+   * Cache for the square root of the number of method calls in a sequence of statements. Once
+   * computed for a given sequence, the value is never updated.
+   */
+  private final Map<Sequence, Double> sequenceMethodSizeSqrt = new HashMap<>();
+
+  /**
+   * Map from a sequence to its approximate execution time in nanoseconds. Once computed for a given
+   * sequence, the execution time of an input sequence is never updated.
+   */
+  private final Map<Sequence, Long> sequenceExecutionTime = new HashMap<>();
+
+  /**
+   * Bias input selection towards lower cost sequences. We first compute and update the weights of
+   * all the candidates within the candidate list before making our selection.
+   *
+   * @param candidates sequences to choose from
+   * @return the chosen sequence
+   */
+  @Override
+  public Sequence selectInputSequence(SimpleList<Sequence> candidates) {
+    double totalWeight = updateWeightMapForCandidates(candidates);
+
+    Sequence selectedSequence = Randomness.randomMemberWeighted(candidates, weightMap, totalWeight);
+    incrementCountInMap(sequenceSelectionCount, selectedSequence);
+    return selectedSequence;
+  }
+
+  /**
+   * Update the weights of the candidates in the given list. If an input sequence has not been
+   * selected or executed before, it will be assigned its default weight, computed by {@code
+   * Sequence.getSize()}. The formula for updating a sequence's weight if selection count and
+   * execution time information are both available is
+   *
+   * <p>1.0 / (k * seq.exec_time * sqrt(seq.meth_size))
+   *
+   * <p>Where k is the number of selections of seq and exec_time is the execution time of seq and
+   * meth_size is the number of method call statements in seq. This formula is a slight
+   * simplification of the one described in the GRT paper which maintains a separate exec_time for
+   * each execution of seq. However, we assume that the execution times of a sequence are the same
+   * as the first execution.
+   *
+   * @param candidates list of candidate sequences
+   * @return the total weight of all the elements in the candidate list
+   */
+  private double updateWeightMapForCandidates(SimpleList<Sequence> candidates) {
+    double totalWeight = 0.0;
+
+    for (int i = 0; i < candidates.size(); i++) {
+      Sequence candidate = candidates.get(i);
+
+      double methodSizeSqrt = getMethodSizeSquareRootForSequence(candidate);
+      Integer selectionCount = sequenceSelectionCount.get(candidate);
+      Long executionTime = sequenceExecutionTime.get(candidate);
+
+      double weight;
+      // Recompute and update this sequence's weight. Note that we check methodSizeSqrt is not zero
+      // to prevent division by zero. This could occur for a sequence that has no method calls.
+      if (selectionCount != null && executionTime != null && methodSizeSqrt != 0) {
```

Comment:
With the changes suggested above, this test should not be necessary, and Orienteering should be self-contained rather than depending on a sequence's natural weight (a concept that doesn't really make sense and we should eliminate).